openstack-mixin/alerts.libsonnet

```diff
@@ -5,75 +5,320 @@
             'for': '5m',
             labels: {
               severity: 'warning',
             },
             annotations: {
               summary: 'VMs are using a high percentage of their allocated memory.',
-              description:
-                (
-                  'Virtual machines on the cloud on {{$labels.instance}} are using {{ printf "%%.0f" $value }} percent of their allocated memory, ' +
-                  'which is above the threshold of %(alertsWarningNovaHighVMMemoryUsage)s percent.'
-                ) % this.config,
+              description: |||
+                Virtual machines on OpenStack {{ $labels.%(instanceFirstLabel)s }} are using {{ printf "%%.0f" $value }} percent of their allocated memory,
+                which is above the threshold of %(alertsWarningNovaHighVMMemoryUsage)s percent.
+              ||| % this.config { instanceFirstLabel: this.config.instanceLabels[0] },
             },
           },
           {
             alert: 'OpenStackNovaHighVMVCPUUsage',
             expr: |||
-              100 * openstack_nova_limits_vcpus_used{%(filteringSelector)s} / clamp_min(openstack_nova_limits_vcpus_max{%(filteringSelector)s}, 1) > %(alertsWarningNovaHighVMVCPUUsage)s
+              100 * openstack_nova_limits_vcpus_used{%(filteringSelector)s} / (openstack_nova_limits_vcpus_max{%(filteringSelector)s} > 0) > %(alertsWarningNovaHighVMVCPUUsage)s
             ||| % this.config,
             'for': '5m',
             labels: {
               severity: 'warning',
             },
             annotations: {
               summary: 'VMs are using a high percentage of their allocated virtual CPUs.',
+              description: |||
+                Virtual machines on OpenStack {{$labels.%(instanceFirstLabel)s}} are using {{ printf "%%.0f" $value }} percent of their allocated virtual CPUs,
+                which is above the threshold of %(alertsWarningNovaHighVMVCPUUsage)s percent.
+              ||| % this.config { instanceFirstLabel: this.config.instanceLabels[0] },
+            },
+          },
+        ],
+      },
+      {
+        name: 'openstack-neutron-alerts' + this.config.uid,
+        rules: [
+          {
+            alert: 'OpenStackNeutronDown',
+            expr: |||
+              openstack_neutron_up{%(filteringSelector)s} == 0
+            ||| % this.config,
+            'for': '5m',
+            labels: {
+              severity: 'critical',
+            },
+            annotations: {
+              summary: 'OpenStack Neutron is down.',
               description:
-                (
-                  'Virtual machines on the cloud on {{$labels.instance}} are using {{ printf "%%.0f" $value }} percent of their allocated virtual CPUs, ' +
-                  'which is above the threshold of %(alertsWarningNovaHighVMVCPUUsage)s percent.'
-                ) % this.config,
+                'OpenStack Neutron service is down on cluster {{ $labels.%(instanceFirstLabel)s }}.'
```

Comment:
Could you explain why you are defining/using this variable instead of just using instance?